### ISSUE TITLE
External Storage shared as subfolder doenst show to (new) users

### ISSUE THREAD

**Issue by g6094199** [nextcloud/server#20094]
### Steps to reproduce
1. mount an external storage
2. create a new usergroup e.g. "cloud" (optional) 
3. share the external storage with the usergroup as a non extisting subfolder of the main folder e.g. /media/subfolder
4. create a new user and put him into the usergroup
5. log in as the new user
6. the folder media is not there
7. create the folder media
8. in the folder media the subfolder appears

similar/relatet to: https://github.com/nextcloud/server/issues/18977

### Expected behaviour
the non existing folder should automatically be created to the users account when step 4 is done

### Server configuration

**Operating system:**Ubuntu 18.04

**Web server:** nginx/1.14.0

**Database:** 10.1.44-MariaDB-0ubuntu0.18.04.1 Ubuntu 18.04

**PHP version:** PHP 7.2.24-0ubuntu0.18.04.3

**Nextcloud version:** Nextcloud 18.0.2

**Updated from an older Nextcloud/ownCloud or fresh install:** update since NC10 or so

**Where did you install Nextcloud from:** 

**Signing status:**
<details>
<summary>Signing status</summary>

No errors have been found.
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - activity: 2.11.0
  - bruteforcesettings: 1.5.0
  - calendar: 2.0.2
  - cloud_federation_api: 1.1.0
  - comments: 1.8.0
  - contacts: 3.2.0
  - dav: 1.14.0
  - federatedfilesharing: 1.8.0
  - federation: 1.8.0
  - files: 1.13.1
  - files_antivirus: 2.2.1
  - files_external: 1.9.0
  - files_external_gdrive: 0.4.0
  - files_pdfviewer: 1.7.0
  - files_rightclick: 0.15.2
  - files_sharing: 1.10.1
  - files_trashbin: 1.8.0
  - files_versions: 1.11.0
  - files_videoplayer: 1.7.0
  - logreader: 2.3.0
  - lookup_server_connector: 1.6.0
  - mail: 1.1.3
  - music: 0.13.0
  - nextcloud_announcements: 1.7.0
  - notes: 3.2.0
  - notifications: 2.6.0
  - oauth2: 1.6.0
  - ocsms: 2.1.7
  - photos: 1.0.0
  - polls: 1.3.0
  - privacy: 1.2.0
  - provisioning_api: 1.8.0
  - recommendations: 0.6.0
  - serverinfo: 1.8.0
  - settings: 1.0.0
  - sharebymail: 1.8.0
  - spreed: 8.0.5
  - support: 1.1.0
  - survey_client: 1.6.0
  - suspicious_login: 3.1.0
  - systemtags: 1.8.0
  - text: 2.0.0
  - theming: 1.9.0
  - twofactor_backupcodes: 1.7.0
  - twofactor_u2f: 5.1.0
  - updatenotification: 1.8.0
  - viewer: 1.2.0
  - workflowengine: 2.0.0

```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "http:\/\/xxxxx.de",
        "trusted_domains": [
            "xxxxxde"
        ],
        "trashbin_retention_obligation": "auto",
        "version": "18.0.2.2",
        "dbtype": "mysql",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "loglevel": 0,
        "maintenance": false,
        "logtimezone": "Europe\/Berlin",
        "updater.release.channel": "beta",
        "theme": "",
        "activity_expire_days": 90,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "mail_smtpmode": "smtp",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpsecure": "tls",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "data-fingerprint": "569fe2a3xxxxxxx388873ecc6",
        "installed": true,
        "app_install_overwrite": [
            "calendar",
            "mail",
            "ocsms",
            "caniupdate",
            "audioplayer",
            "files_external_gdrive",
            "polls",
            "spreed",
            "contacts",
            "bruteforcesettings",
            "files_rightclick",
            "talk_simple_poll"
        ],
        "mysql.utf8mb4": true
    }
}

```
</details>

**Are you using external storage, if yes which one:** local (via sshfs)/google/webdav/federation

**Are you using encryption:** no

**Comment by llucax** [comment]
I can confirm this. I didn't figured out mounting in a sub-folder didn't work because I haven't created the folder until I saw this issue. I was thinking mounting in subfolders was not supported, but then how is it that the mounted folder couldn't be seen except if I went to the External Storage in the web UI. That also looked like a feature, to not have a mount point in the fs at all and have to access via External Storage.

Anyway, it would be good any missing folders are automatically created for users that don't have them but have access to the external storage.
